graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java

```diff
@@ -61,82 +65,71 @@ public void validate(SyntaxNodeAnalysisContext context) {
             }
         }
     }
 
-    private void validateReturnType(FunctionDefinitionNode functionDefinitionNode, SyntaxNodeAnalysisContext context) {
-        MethodSymbol methodSymbol = PluginUtils.getMethodSymbol(context, functionDefinitionNode);
-        if (methodSymbol != null) {
-            Optional<TypeSymbol> returnTypeDesc = methodSymbol.typeDescriptor().returnTypeDescriptor();
-            if (returnTypeDesc.isPresent()) {
-                // if return type is a union
-                if (returnTypeDesc.get().typeKind() == TypeDescKind.UNION) {
-                    validateReturnTypeUnion(context,
-                            ((UnionTypeSymbol) returnTypeDesc.get()).memberTypeDescriptors(), functionDefinitionNode);
+    private void validateReturnType(MethodSymbol methodSymbol,
+                                    Location location, SyntaxNodeAnalysisContext context) {
+        Optional<TypeSymbol> returnTypeDesc = methodSymbol.typeDescriptor().returnTypeDescriptor();
+        if (returnTypeDesc.isPresent()) {
+            // if return type is a union
+            if (returnTypeDesc.get().typeKind() == TypeDescKind.UNION) {
+                validateReturnTypeUnion(context,
+                        ((UnionTypeSymbol) returnTypeDesc.get()).memberTypeDescriptors(), location);
+            } else {
+                // if return type not a union
+                // nil alone is invalid - must have a return type
+                if (returnTypeDesc.get().typeKind() == TypeDescKind.NIL) {
+                    PluginUtils.updateContext(context, CompilationErrors.INVALID_RETURN_TYPE_NIL, location);
+                } else if (returnTypeDesc.get().typeKind() == TypeDescKind.ERROR) {
+                    PluginUtils.updateContext(context, CompilationErrors.INVALID_RETURN_TYPE_ERROR, location);
                 } else {
-                    // if return type not a union
-                    // nil alone is invalid - must have a return type
-                    if (returnTypeDesc.get().typeKind() == TypeDescKind.NIL) {
-                        PluginUtils.updateContext(context, CompilationErrors.INVALID_RETURN_TYPE_NIL,
-                                                  functionDefinitionNode.location());
-                    } else if (returnTypeDesc.get().typeKind() == TypeDescKind.ERROR) {
-                        PluginUtils.updateContext(context, CompilationErrors.INVALID_RETURN_TYPE_ERROR,
-                                                  functionDefinitionNode.location());
-                    } else {
-                        if (hasInvalidReturnType(returnTypeDesc.get())) {
-                            PluginUtils.updateContext(context, CompilationErrors.INVALID_RETURN_TYPE,
-                                                      functionDefinitionNode.location());
-                        }
+                    if (hasInvalidReturnType(returnTypeDesc.get(), location, context)) {
```

Comment:
Can't we move this `if` condition to the upper level, instead of the `else` part?

Reply:
Hope this function is clearer now. Fixed in 06ffce570ab4dd469fcd83dea3784ac42114339b